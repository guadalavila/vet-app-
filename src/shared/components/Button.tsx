import React from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, ViewStyle } from 'react-native';
import CustomText from './CustomText';
import { colors } from '../utils/colors';
import { size } from '../utils/size';
import { typography } from '../utils/typography';

interface IButtonProps {
    title: string;
    onPress: () => void;
    style?: StyleProp<ViewStyle> | undefined;
}

const Button = ({ title, onPress, style = {} }: IButtonProps) => {
    return (
        <TouchableOpacity style={[styles.button, style]} activeOpacity={0.7} onPress={onPress}>
            <CustomText style={[styles.text]}>{title}</CustomText>
        </TouchableOpacity>
    );
};

export default Button;

const styles = StyleSheet.create({
    button: {
        borderRadius: 10,
        marginHorizontal: size.XXL,
        paddingHorizontal: size.XL,
        paddingVertical: size.XXL,
        marginVertical: size.XL,
        backgroundColor: colors.light.primary,
    },
    text: {
        fontWeight: '700',
        fontSize: typography.size.S,
        textAlign: 'center',
        color: colors.light.white,
    },
});
