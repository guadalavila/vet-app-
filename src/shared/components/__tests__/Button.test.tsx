import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import Button from '../Button';

describe('Button', () => {
    const onPressMock = jest.fn();

    test('should match snapshot', () => {
        const container = render(<Button title='Press me' onPress={onPressMock} />);
        expect(container).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<Button title='Press me' onPress={onPressMock} />);
        expect(screen.getByTestId('button')).toBeTruthy();
    });

    test('should render correct text', () => {
        render(<Button title='Press me' onPress={onPressMock} />);
        expect(screen.getByText('Press me')).toBeTruthy();
    });

    test('should call a function when press button', () => {
        render(<Button title='Press me' onPress={onPressMock} />);
        const button = screen.getByTestId('button');
        fireEvent.press(button);
        expect(onPressMock).toHaveBeenCalled();
    });
});
