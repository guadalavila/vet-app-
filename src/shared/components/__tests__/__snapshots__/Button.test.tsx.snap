// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#A765C1",
      "borderRadius": 10,
      "marginHorizontal": 14,
      "marginVertical": 10,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 14,
    }
  }
  testID="button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
        },
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          },
          Object {
            "color": "#F1F1F1",
          },
        ],
      ]
    }
    testID="button-text"
  >
    Press me
  </Text>
</View>
`;
